engine/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.services
+
+import akka.actor.{Actor, Props}
+import com.typesafe.config.Config
+import cromwell.backend.BackendJobDescriptorKey
+import cromwell.core.WorkflowId
+import cromwell.services.MetadataServiceActor._
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import org.joda.time.DateTime
+
+import scala.language.postfixOps
+
+object MetadataServiceActor {
+
+  final val MetadataServiceName = "MetadataService"
+
+  case class MetadataJobKey(callFqn: String, index: Option[Int], attempt: Int)
+  object MetadataJobKey {
+    def apply(backendJobKey: BackendJobDescriptorKey): MetadataJobKey = MetadataJobKey(backendJobKey.call.fullyQualifiedName, backendJobKey.index, backendJobKey.attempt)
+  }
+
+  case class MetadataKey(workflowId: WorkflowId, jobKey: Option[MetadataJobKey], key: String)
+  case class MetadataValue(value: String, timestamp: DateTime)
+  object MetadataValue {
+    def apply(value: String): MetadataValue = MetadataValue(value, DateTime.now)
+  }
+
+  case class MetadataEvent(key: MetadataKey, value: MetadataValue)
```

Comment:
Agree that time seems to be more affiliated with the event than the value, if that's what you were suggesting. OTOH if I came across that already changed I'd probably bellyache about having the value being stringly typed ;)